lib/index.js

```diff
@@ -22,36 +25,49 @@ const createRule = (name, browserstring, description, { ts = null } = {}) => {
     ),
     create(context) {
       let browsers = browserslist(browserstring);
-      const config = findConfig(path.dirname(context.getFilename())) || {
+      const config = findConfig(path.dirname(
+        context.filename ?? context.getFilename()
+      )) || {
         defaults: "defaults",
       };
       const desiredBrowsers = browserslist(config.defaults);
       const badBrowsers = desiredBrowsers
-        .filter((browser) => browsers.indexOf(browser) !== -1)
+        .filter((browser) => browsers.includes(browser))
         .join(", ");
-      if (badBrowsers) {
+      if (globalThis.ESLINT_TESTING || badBrowsers) {
```

Comment:
The ESLint RuleTester now fails in ESLint 9 if supplied a rule-as-function (since such rules are no longer allowed), so we could no longer directly require the rule file. And since I switched to requiring the index file instead, browsers would end up being checked (the defaults specifically) and if not found to be problematic, the rule would not run at all, causing problems with tests which expected to fail not failing.